core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -67,12 +69,13 @@ public SpoolingExchangeOutputBuffer(
     public OutputBufferInfo getInfo()
     {
         BufferState state = stateMachine.getState();
+        LocalMemoryContext memoryContext = getSystemMemoryContextOrNull();
```

Comment:
nit: would be more type safe to return `Optional<LocalMemoryContext>` instead

Reply:
This method is used in the `updateMemoryUsage` method that is called on every `enqueue`. Returning an `Optional` would result in extra object allocation on every `enqueue` call. I'm not sure if it's worth it since it is a private method.